### ISSUE TITLE
College Algebra with Python - Incorrect assignment statement can pass the test

### ISSUE THREAD

**Issue by mmatsumoto1026** [freeCodeCamp/freeCodeCamp#50607]
### Describe the Issue

In College Algebra with Python, the test of "Build a Multi-Function Calculator" Certification project's Step 18 - Finding Square Factors seems to be able to pass the incorrect assignment statement ~~which cause that print() function will print the number which is not the greatest square factor of the camper-entered number~~.

By the code shown below, after the camper enters the number `32`, print() function's argument `max_factor**2` will become `256` and this is not the greatest square factor of entered number, but ~~the function will print this and~~ the test will pass.

### Affected Page

https://colab.research.google.com/drive/1jT7atuRCOnkdPyDmlgKVJqxouDSx7Ioh?usp=sharing

### Your code

The code is from Step 18 in notebook.

```python
import math

n = int(input('Enter an integer to find the greatest square factor: '))

max_factor = 1
upper_limit = math.floor(math.sqrt(n)) + 1

# Change one line in this loop:
for maybe_factor in range(1,upper_limit):
    if n % (maybe_factor**2) == 0:
        max_factor = maybe_factor**2

# Keep this print statement:
print(max_factor**2)
```

### Expected behavior

The test should detect the incorrect assinment statement ~~and thereby avoid printing anything that is not the greatest square factor of the entered number~~ and avoid passing it so that the camper can aware why the number which is not the correct greatest square factor is shown.

### Screenshots

![Incorrect_assinment_statement_can_pass_the_test](https://github.com/freeCodeCamp/freeCodeCamp/assets/44451585/d2e54259-eba2-4ede-863c-e21e5795ec95)

### System

- Device: Laptop
- OS: Windows 11 Home
- Browser: Chrome
- Version: Version 114.0.5735.91

**Comment by midikaif** [comment]
@mmatsumoto1026 answer should be 16?

**Comment by midikaif** [comment]
i am unable to find the affected page in repo of freecodecamp.

**Comment by midikaif** [comment]
will anyone help?

**Comment by Dario-DC** [comment]
Thank you for opening this issue. As only staff has edit access to the Colab Notebooks this will be evaluated and fixed internally.

cc/ @edatfreecodecamp

**Comment by mmatsumoto1026** [comment]
Thanks for your comments, @midikaif and @Dario-DC.

> answer should be 16?

Yes, the greatest square factor of 32 is 16. 

But I'm not sure whether the correct greatest square factor (e.g. 16) of the number camper entered (e.g. 32) should always be displayed after running the program. (I had not been able to realize this when I open this issue, so I fixed the issue's description later.)
It seems that Campers might be able to notice that the program they wrote in the loop is incorrect when a different number is displayed than expected by the print() function after entering the number.

**Comment by edatfreecodecamp** [comment]
I just fixed the issue

**Comment by mmatsumoto1026** [comment]
@edatfreecodecamp Thanks for your amendment.

After confirming [the fix commitment](https://github.com/edatfreecodecamp/python-math/commit/459609e18c9196234d68f8b2f1de298e408a5bb8) in your repository, I realized that I had made a mistake in my issue description which is ambiguous in explaining the cause.

The reason why the test passes even though the greatest square factor is not displayed for the number entered by the camper (e.g. 32) seems to be that the test program in step 18 of [the test library](https://raw.githubusercontent.com/freeCodeCamp/cdn/main/build/math-cert-tests/math-code-test-a.py) that is first imported in Google Colab does not have a dot('.') immediately after `maybe_factor` to detect unnecessary characters, which is different from the code before you modified it in latest commitment (I believe that the judgment condition including the dot at the end of the regex is correct after confirming the test condition with the dot exactly detect the incorrect answer in my local environment).

Therefore, it seems that the way to resolve the issue was to update the contents of [the test program](https://raw.githubusercontent.com/freeCodeCamp/cdn/main/build/math-cert-tests/math-code-test-a.py) that is first imported into Google Colab (but only to update the test program before the latest fix commitment was made), even though the `max_factor = maybe_factor**2` used in the issue description was an incorrect answer and the test passed despite that.
